File: manual/code/generate_strategies/IR(summary)/java/java/function_Schematics_basetype_validate.java
package com.example;

import com.google.common.primitives.Ints;

public class PositiveIntegerType {
    
    public String validate(String value) {
        
        Integer intValue = Ints.tryParse(value);
        if (intValue == null) {
            
            return "Value must be an integer.";
        }
        
        if (intValue <= 0) {
            return "Value must be a positive integer.";
        }
        
        return "Validation successful.";
    }
}

File: manual/code/generate_strategies/IR(pseudocode)/python/python/function_Schematics_basetype_validate.py
from schematics.types import BaseType

class PositiveIntegerType(BaseType):

    def validate(self, value: str) -> str:
        try:
            value = int(value) 
        except ValueError:
            return "Value must be an integer."
        
        if value <= 0:
            return "Value must be a positive integer."
        
        return "Validation successful."
